plugin-process-to-cgroup-bridge/src/tests.rs

```diff
@@ -0,0 +1,437 @@
+        );
+
+        let agent = agent::Builder::new(plugins)
+            .with_expectations(runtime_expectations)
+            .build_and_start()
+            .unwrap();
+
+        agent.wait_for_shutdown(Duration::from_secs(2)).unwrap();
+
+        Ok(())
+    }
+
+    fn prepare_procfs_mock() -> anyhow::Result<PathBuf> {
+        let tmp = tempdir()?;
+        let base_path = tmp.keep();
+
+        let entries = [
+            Entry {
+                path: "1",
+                entry_type: EntryType::Dir,
+            },
+            Entry {
+                path: "2",
+                entry_type: EntryType::Dir,
+            },
+            Entry {
+                path: "3",
+                entry_type: EntryType::Dir,
+            },
+            Entry {
+                path: "1/cgroup",
+                entry_type: EntryType::File("0::/system.slice/shared.slice"),
+            },
+            Entry {
+                path: "2/cgroup",
+                entry_type: EntryType::File("0::/system.slice/shared.slice"),
+            },
+            Entry {
+                path: "3/cgroup",
+                entry_type: EntryType::File("0::/system.slice/single.slice"),
+            },
+        ];
+
+        create_mock_layout(base_path.clone(), &entries)?;
+        Ok(base_path)
+    }
+
+    fn prepare_mock_measurements(ctx: &mut TransformCheckInputContext) -> anyhow::Result<MeasurementBuffer> {
+        let metric_a = ctx
+            .metrics()
+            .by_name("metric_a")
+            .expect("metric_a metric should exist")
+            .0;
+        let metric_b = ctx
+            .metrics()
+            .by_name("metric_b")
+            .expect("metric_b metric should exist")
+            .0;
+
+        let mut m = MeasurementBuffer::new();
+
+        let create_point = |timestamp, metric, consumer, value| {
+            MeasurementPoint::new_untyped(
+                timestamp,
+                metric,
+                Resource::LocalMachine,
+                consumer,
+                WrappedMeasurementValue::U64(value),
+            )
+        };
+
+        // metric_a points
+        m.push(create_point(*T, metric_a, ResourceConsumer::Process { pid: 1 }, 10));
+        m.push(create_point(*T2, metric_a, ResourceConsumer::Process { pid: 1 }, 10));
+        m.push(create_point(*T, metric_a, ResourceConsumer::Process { pid: 2 }, 10));
+        m.push(create_point(*T, metric_a, ResourceConsumer::Process { pid: 3 }, 10));
+        m.push(create_point(*T, metric_a, ResourceConsumer::LocalMachine, 10));
+
+        // metric_b points
+        m.push(create_point(*T, metric_b, ResourceConsumer::Process { pid: 1 }, 10));
+        m.push(create_point(*T, metric_b, ResourceConsumer::Process { pid: 2 }, 10));
+
+        Ok(m)
+    }
+
+    fn config_to_toml_table(config: &Config) -> toml::Table {
+        toml::Value::try_from(config).unwrap().as_table().unwrap().clone()
+    }
+
+    // Mock filesystem utilities
+    enum EntryType<'a> {
```

Comment:
You could use `&'static str` in these strucs to avoid the lifetime parameter. It would work since you are using constant strings in your test. But it's fine the way it is too.